### ISSUE TITLE
[HOLD for payment 2024-08-14] [$250] iOS - Expensify Card - Content on Expensify Card intro screen is truncated

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#45901]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.10-2
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Issue reported by:** Applause - Internal Team

## Action Performed:
Precondition:
- Device font size is increased
1. Launch New Expensify app
2. Go to workspace settings
3. Go to Expensify Card (enable on More features if not yet)

## Expected Result:
The content will be displayed in two lines if it does not fit the screen (similar behavior with options on onboarding screen)

## Actual Result:
"Up to 2% cash back on every US purchase" and "Spend controls and custom limits" are truncated

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [x] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/user-attachments/assets/a33c34ae-d848-4757-978b-27c268abf583

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0126b472d9d6b81559</li>
        <li>Upwork Job ID: 1815823007102686005</li>
        <li>Last Price Increase: 2024-07-23</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @slafortune</details>

**Comment by lanitochka17** [comment]
@slafortune  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by etCoderDysto** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Expensify Card - Content on Expensify Card intro screen is truncated
### What is the root cause of that problem?
We are not passing any value to `numberOfLinesTitle` prop of `MenuItem`. And `MenuItem` will use the default `numberOfLinesTitle = 1` and truncates the contents.  
https://github.com/Expensify/App/blob/e3ebeffb55c2a20ee890061f23e3e0f37d32290e/src/components/FeatureList.tsx#L101-L108
### What changes do you think we should make in order to solve the problem?
We should pass `numberOfLinesTitle={0}` to the MenuItem to override the default value. No truncation will be applied on the content when the value is 0.
### What alternative solutions did you explore? (Optional)
We can set 'numberOfLinesTitle' to 2 of we don't want the number of lines not to exceed above 2 lines

**Comment by slafortune** [comment]
With the font size increased, do we still expect to see full information or is truncated lines the expected outcome?

**Comment by dubielzyk-expensify** [comment]
I don't think we should truncate the text given users should still be able to read it. So I expect two lines of text in this instance. 

cc @Expensify/design for extra thoughts.

**Comment by shawnborton** [comment]
Yup, I agree that the text shouldn't be truncated.

**Comment by dannymcclain** [comment]
Definitely agree.

**Comment by etCoderDysto** [comment]
**[Proposal updated](https://github.com/Expensify/App/issues/45901#issuecomment-2242846656)**
* I have added screenshot of the result

**Comment by Pujan92** [comment]
Bump @MonilBhavsar for the review of https://github.com/Expensify/App/issues/45901#issuecomment-2246214018

**Comment by etCoderDysto** [comment]
@Pujan92 @MonilBhavsar Because of changes made in [this PR ](https://github.com/Expensify/App/pull/45690), we are now displaying a new empty state modal in Expensify card page. And we are no longer displaying [FeatureList](https://github.com/Expensify/App/blob/8942513f6eab949d669983d6260ba705a3f894e9/src/components/FeatureList.tsx#L101). The only places we are using FeatureList is on BookTravel page and empty workspace list page. And the copy on those pages are very short. I don't think they will be truncated. 

**Expensify Card Page**
<img width="404" alt="Screenshot 2024-07-31 at 12 12 29 at night" src="https://github.com/user-attachments/assets/8e64466e-fc19-4ee7-88a5-940f519de84f">

**Book travel page**
<img width="404" alt="image" src="https://github.com/user-attachments/assets/bbcd120f-e8aa-4465-9f26-ae22a21eab90">

**Empty workspace list page**
<img width="404" alt="image" src="https://github.com/user-attachments/assets/a6c57d41-c170-4af7-8e95-73a0078f817e">

The screenshots are taken with an increased fonts as the test step claims. _**Considering this new changes, should we pursue fixing the issue?**_

**Comment by shawnborton** [comment]
> Because of changes made in https://github.com/Expensify/App/pull/45690, we are now displaying a new empty state modal in Expensify card page.

To be clear, we first show the empty state style page with the big illustration and features below. Then once the feature is enabled, you see the empty state with the rows in the background. So this page uses both of the empty states depending on the state of the feature.

**Comment by Pujan92** [comment]
Seems an issue from this [commit](https://github.com/Expensify/App/pull/45690/commits/8c91902652ae728e9cf5594690fd6f565c47a0d9), -1 for `paymentBankAccountID` will be considered as truthy and due to that it renders the incorrect content. 

https://github.com/Expensify/App/blob/864c69f61e03669c20f2e0fcddd99f0291fb2240/src/pages/workspace/expensifyCard/WorkspaceExpensifyCardPage.tsx#L27

cc: @koko57 

@etCoderDysto you can adjust to render `WorkspaceExpensifyCardPageEmptyState` and raise the PR.

**Comment by koko57** [comment]
@Pujan92 yes it can be changed to 0

**Comment by shah-pujan** [comment]
Thanks @koko57, @etCoderDysto Let's do that.

**Comment by Pujan92** [comment]
Thanks @koko57, @etCoderDysto Let's do that.

**Comment by etCoderDysto** [comment]
Great! I will update the value to 0 @Pujan92

**Comment by etCoderDysto** [comment]
> To be clear, we first show the empty state style page with the big illustration and features below. Then once the feature is enabled, you see the empty state with the rows in the background. So this page uses both of the empty states depending on the state of the feature.

@shawnborton can you please confirm if I get the expected result right? 
1. When user navigates to Expensify card they will see the features list with illustration 
2. When they select a bank account they will see empty states modal 

https://github.com/user-attachments/assets/24979bf3-c496-44d2-889d-101bf474cd77

**Comment by shawnborton** [comment]
Hmm let's see what @joekaufmanexpensify says. But I think what you are showing is correct?

**Comment by shawnborton** [comment]
Regardless though, I don't think that conversation relates to the original bug being reported so let's try not to distract from the original bug we need to fix.

**Comment by etCoderDysto** [comment]
> Regardless though, I don't think that conversation relates to the original bug being reported so let's try not to distract from the original bug we need to fix.

Currently we are not displaying feature list page before the user selects bank account as this [PR](https://github.com/Expensify/App/pull/45690) mentions. Therefore we couldn't see the fix for the original report unless the issue that the user is not seeing feature list page before selecting bank account is fixed. 

Here is the existing behaviour. You couldn't see feature list before you select bank account

https://github.com/user-attachments/assets/ddcaaf66-6639-4abf-9e90-ede6d74308f3




cc: @koko57

**Comment by joekaufmanexpensify** [comment]
>Hmm let's see what @joekaufmanexpensify says. But I think what you are showing is correct?

Is what you're trying to confirm whether it is correct that after clicking `Issue card` and selecting a bank account, that the empty state for the card feed changes, or that you immediately begin the flow to issue a card, or both?

**Comment by etCoderDysto** [comment]
> > Hmm let's see what @joekaufmanexpensify says. But I think what you are showing is correct?
> 
> Is what you're trying to confirm whether it is correct that after clicking `Issue card` and selecting a bank account, that the empty state for the card feed changes, or that you immediately begin the flow to issue a card, or both?

I wanted to confirm both. Issue new card with Get Expensify card illustration -> Add Bank -> Empty state card feed. Think this the correct flow. You can see the flow [here](https://github.com/Expensify/App/issues/45901#issuecomment-2265147097)

**Comment by joekaufmanexpensify** [comment]
Got it. It is definitely correct that the empty state changes after selecting a bank account as shown in your video. However, it's not clear to me based on the (internal) design doc whether we're supposed to go directly into issuing a card after selecting a bank account, or land back on the empty state, and click the `Issue card` button in the top right. 

The [design doc is clear](https://docs.google.com/document/d/1sTg-e11jdOmregOFBhdO8ALlykKl8PqTeTEAgEMS6mU/edit#bookmark=id.67wffhjy6vqe) that if you're adding a VBA from scratch, you land back on `Company cards` and then need to click `Issue card` in the top right to issue a card. But I don't see that it covers whether we jump right into issuing a card if you're just selecting an existing bank account from the list. 

@kevinksullivan mind weighing in on the intention for this case when you wrote the doc?

**Comment by slafortune** [comment]
@Pujan92 can you please complete the check list?

**Comment by slafortune** [comment]
@etCoderDysto can you please share your Upworks profile - I'm having a hard time finding you.

**Comment by etCoderDysto** [comment]
> @etCoderDysto can you please share your Upworks profile - I'm having a hard time finding you.

Sorry! Here is a link to my profile: https://www.upwork.com/freelancers/~01ea78e0164390eb79

**Comment by slafortune** [comment]
Offer sent - https://www.upwork.com/nx/wm/offer/103525438

**Comment by etCoderDysto** [comment]
I have accepted the offer. Thanks!

**Comment by slafortune** [comment]
bump @Pujan92,  can you please complete the check list?

**Comment by Pujan92** [comment]
**BugZero Checklist**: The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [X] [@Pujan92] The PR that introduced the bug has been identified. Link to the PR: https://github.com/Expensify/App/pull/44770
- [X] [@Pujan92] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: https://github.com/Expensify/App/pull/44770/files#r1720206013
 - [X] [@Pujan92] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: **NA**
 - [X] [@Pujan92] Determine if we should create a regression test for this bug. **Yes**
 - [X] [@Pujan92] If we decide to create a [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) for the bug, please propose the regression test steps to ensure the same bug will not reach production again.
 
### Regression Test Steps

1. Goto workspace settings page
2. Enable Expensify Card from More features if it is not enabled yet
3. Goto the Expensify Card item
4. Verify the features text points aren't truncated

**Comment by JmillsExpensify** [comment]
$250 approved for @Pujan92
